### PR TITLE
[mlir] Fix missing FuncOps.h.inc

### LINKED ISSUE #84568: 18.1.0 build from source missing mlir/Dialect/Func/IR/FuncOps.h.inc?
I download llvm-project-18.1.0.src.tar.xz and tried to build with the following:

```
[]$ cmake                                                            \
? -DLLVM_TARGETS_TO_BUILD=X86                                        \
? -DCLANG_CONFIG_FILE_USER_DIR="${SRC}/${VER}/${TAR}/llvm.cfg"       \
? -DCMAKE_C_COMPILER=${HOST_GCC}/bin/gcc                             \
? -DCMAKE_CXX_COMPILER=${HOST_GCC}/bin/g++                           \
? -DCMAKE_BUILD_TYPE="Release"                                       \
? -DLIBOMP_ARCH="x86_64"                                             \
? -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}                                       \
? -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;cross-project-tests;libc;lld;lldb;mlir;polly;pstl" \
? -DLLVM_ENABLE_RUNTIMES="compiler-rt;libc;libcxx;libcxxabi;libunwind;openmp"    \
? -DCMAKE_CXX_LINK_FLAGS="-L${HOST_GCC}/lib64 -Wl,-rpath,${HOST_GCC}/lib64"      \
? ../llvm
  .
  .
  .
[]$ make
  .
  .
  .
[ 71%] Built target MLIRSCFDialect
[ 71%] Building CXX object tools/mlir/lib/Dialect/Affine/Analysis/CMakeFiles/obj.MLIRAffineAnalysis.dir/AffineAnalysis.cpp.o
In file included from /scr/LLVM/llvm-18.1.0/llvm-project-18.1.0.src/mlir/lib/Dialect/Affine/Analysis/AffineAnalysis.cpp:20:
/scr/LLVM/llvm-18.1.0/llvm-project-18.1.0.src/mlir/include/mlir/Dialect/Func/IR/FuncOps.h:29:10: fatal error: mlir/Dialect/Func/IR/FuncOps.h.inc: No such file or directory
   29 | #include "mlir/Dialect/Func/IR/FuncOps.h.inc"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

I'm not sure what the appropriate work around is.  Maybe just remove mlir from projects?

### PR DESCRIPTION
**Pull request by jschueller**
Closes #84568

Similar to #71691 (reproduce with make -j1)

 /cc @tblah

### CODE REVIEW

**Review by joker-eph — APPROVED** [review]

**Review by tblah — APPROVED** [review]

**Comment by jschueller** [comment]
hello @tblah, maybe this could be merged ?

**Comment by tblah** [comment]
> hello @tblah, maybe this could be merged ?

Yeah this LGTM. Do you need help merging?

**Comment by jschueller** [comment]
yes, I dont have write access, could you please merge it ?

File: mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,9 @@ add_mlir_dialect_library(MLIRAffineAnalysis
   ADDITIONAL_HEADER_DIRS
   ${MLIR_MAIN_INCLUDE_DIR}/mlir/Dialect/Affine
 
+  DEPENDS
+  MLIRFuncOpsIncGen
```
Comment (joker-eph) [inline mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt:12]: This does not seem like the right dependency to me: we should depend on the FuncDialect here. The IncGen targets are meant to be depending on by the dialects themselves.

You can also see this as a transitive dependency issue: you're skipping one step in the dependency chain.

**Pull request merged**